src/main/java/Duke.java

```diff
@@ -1,10 +1,172 @@
+                        throw new IllegalArgumentException("looks like the description is missing");
+                    }
+                    tasks.add(new Todo(description));
+                    System.out.println("Got it. I've added this task:");
+                    System.out.println("  " + tasks.get(tasks.size() - 1));
+                    System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+                
+                } else if (userInput.startsWith("deadline")) {
+                    if (userInput.length() <= 9) {
+                        throw new IllegalArgumentException("looks like something is missing (description/ Deadline)");
+                    }
+                    String[] parts = userInput.substring(9).split("/by");
+                    String description = parts[0].trim();
+                    String by = parts[1].trim();
+                    if (description.isEmpty() || by.isEmpty()) {
+                        throw new IllegalArgumentException("looks like something is missing (description/ Deadline)");
+                    }
+                    tasks.add(new Deadline(description, by));
+                    System.out.println("Got it. I've added this task:");
+                    System.out.println("  " + tasks.get(tasks.size() - 1));
+                    System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+
+                } else if (userInput.startsWith("event")) {
+                    if (userInput.length() <= 6) {
+                        throw new IllegalArgumentException("looks like something is missing (description/ Deadline)");
+                    }
+                    String[] parts = userInput.substring(6).split("/from");
+                    String description = parts[0].trim();
+                    String[] eventParts = parts[1].trim().split("/to");
+                    String from = eventParts[0].trim();
+                    String to = eventParts[1].trim();
+                    if (description.isEmpty() || from.isEmpty() || to.isEmpty()) {
+                        throw new IllegalArgumentException(
+                                "looks like something is missing (description/ start date/ end date)");
+                    }
+                    tasks.add(new Event(description, from, to));
+                    System.out.println("Got it. I've added this task:");
+                    System.out.println("  " + tasks.get(tasks.size() - 1));
+                    System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+
+                } else if (userInput.startsWith("mark")) {
+                    System.out.println(" Nice! I've marked this task as done:");
+                    int taskNumber = Integer.parseInt(userInput.split(" ")[1]);
+                    tasks.get(taskNumber - 1).markAsDone();
+
+                    for (int i = 0; i < tasks.size(); i++) {
+                        System.out.println((i + 1) + ". " + tasks.get(i));
```

Comment:
I liked how you have added white spaces within the statements according to the [Java Coding Standard](https://se-education.org/guides/conventions/java/index.html).